lib/cli-engine.js

```diff
@@ -459,6 +474,30 @@ class CLIEngine {
              */
             this._lintResultCache = new LintResultCache(cacheFile, this.config);
         }
+
+        // setup special filter for fixes
+        if (this.options.fix && this.options.fixTypes && this.options.fixTypes.length > 0) {
+
+            // throw an error if any invalid fix types are found
+            validateFixTypes(this.options.fixTypes);
+
+            // convert to Set for faster lookup
+            const fixTypes = new Set(this.options.fixTypes);
+
+            // save original value of options.fix in case it's a function
+            const originalFix = this.options.fix;
+
+            // create a cache of rules (but don't populate until needed)
+            this._rulesCache = null;
+
+            this.options.fix = lintResult => {
+                const rule = this._rulesCache.get(lintResult.ruleId);
+                const matches = rule.meta && fixTypes.has(rule.meta.type);
+
+                return matches && (typeof originalFix === "function" ? originalFix(lintResult) : true);
```

Comment:
I'm not sure this would make much of a difference between `typeof` just does a value lookup, which is pretty fast. That said, I'll still make the suggested change in the spirit of keeping things DRY.